### PR TITLE
[ATH-1036] Fix ath-container and docker based tests

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
See https://github.com/jenkinsci/acceptance-test-harness/issues/1036 since #878 docker based tests are not run on ci.jenkins.io because `docker` can not be found on the `ath` image. 

I have updated the docker installation process according to https://docs.docker.com/engine/install/ubuntu/ and also reverted https://github.com/jenkinsci/acceptance-test-harness/commit/c54ef5f8f175ef37d1cd9b552755bfba85b12247#diff-59086a3e54e38095246aa3a051f9020274dbf005b4652a7808cd4c6c239c536eL67 so `ath-user` can see and run the docker command.

I have checked that the image builds and you can run docker inside as `ath-user`. Note that by running again docker tests we may find failures that were ignored before by accident. The purpose of this PR is not to solve those tests but to make the failures visible again. To check this is working we can compare the number of tests run for the `WorkflowPluginTest` suite in `master` and for this PR  

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes

### CODE REVIEW

File: src/main/resources/ath-container/Dockerfile
Comment (jtnord) [inline src/main/resources/ath-container/Dockerfile:26]: So.. isn't this a case of just using Ng the right package in Ubuntu?
https://packages.ubuntu.com/jammy/amd64/docker.io/filelist

If we wanted the latest docker-ce then the scripts are fine, but I think we can still use apt can't we?

Comment (dduportal) [reply src/main/resources/ath-container/Dockerfile:26]: The « repackaged » docker for ubuntu had been a pain for years on the productions we managed. It kinda shift quite often from official docker ce (not mentioning missing buildx, time to have cve fixed etc.). 
Using the docker-ce package is « only » adding a few lines here, but for the better.

Comment (raul-arabaolaza) [reply src/main/resources/ath-container/Dockerfile:26]: And in many cases you get a too old version of docker. Using the official repos from docker makes it easier to update versions if needed plus contains the most recent versions and fixes if needed.

**Comment by raul-arabaolaza** [comment]
Too many failures here, and I can not make sense of the error, I suspect it is because of an infra problem.

On the good side it looks like more tests are run, I will retrigger this build by using an empty commit

File: src/main/resources/ath-container/Dockerfile
Comment (jtnord) [inline src/main/resources/ath-container/Dockerfile:67]:
```suggestion
      $(lsb_release -cs) stable" > /etc/apt/sources.list.d/docker.list
```

the original doc has `| sudo tee > dev/null` because the useris not running as root / has no permissions on `/etc/apt/sources.list`  so running tee as root means it ca write to the file and std.out (and std.out is thrown away) - we can just redirect std.out to the file.

**Review by jtnord — APPROVED** [review]

**Comment by jtnord** [comment]
it looks like something has seriously regressed in either the LTS or was merged that broke pretty much all the tests.

https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/master/683/testReport/

could not see any merges around this timeline and is mostly around save - given the nature of the first few tests I looked at...

**Comment by raul-arabaolaza** [comment]
So, 56 skipped tests in this pr vs 226 in master -> 170 new tests run

Also for example https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/PR-1037/4/testReport/plugins/WorkflowPluginTest/java_11_jenkins_latest_split6___hello_world_from_git/ is run in this pr, but in https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/master/684/testReport/plugins/WorkflowPluginTest/java_11_jenkins_lts_split8___hello_world_from_git/ is skipped because docker is not found.

I do believe we can merge this, as failures are unrelated and the goal of unlocking docker tests is achieved

**Review by timja — CHANGES_REQUESTED** [review]
> I do believe we can merge this, as failures are unrelated and the goal of unlocking docker tests is achieved

Please do not merge this, a broken main build is the most important activity. Without a green baseline we cannot correctly evaluate this change and make sure there's no regression introduced, e.g. if this breaks any docker tests etc.

**Comment by timja** [comment]
> it looks like something has seriously regressed in either the LTS or was merged that broke pretty much all the tests.

Filed https://github.com/jenkinsci/acceptance-test-harness/issues/1039 which should be fixed before merging this.
From an initial look so far seem to be infra / container related, on my local machine the test I ran passed just fine.

**Comment by raul-arabaolaza** [comment]
I do not have more time to dedicate to this for the moment. If anyone wants to use this as base for future PRs please go forward.

**Comment by timja** [comment]
Seems lots of docker tests are failing for not being able to build the image currently

**Comment by raul-arabaolaza** [comment]
Let me check, I had problems with old docker versions locally

**Comment by raul-arabaolaza** [comment]
Not all tests are failing because of that, of my random 4 picks it was 50%, anyway, let's see what is failing

**Comment by raul-arabaolaza** [comment]
Oh, ``` 2.977 Package viewvc is not available, but is referred to by another package.
#6 2.977 This may mean that the package is missing, has been obsoleted, or
#6 2.977 is only available from another source
#6 2.977 
#6 3.040 E: Package 'viewvc' has no installation candidate```

Those kind of errors should be easy to fix or to die trying

**Comment by raul-arabaolaza** [comment]
https://github.com/viewvc/viewvc/issues/310

**Comment by raul-arabaolaza** [comment]
https://community.developer.atlassian.com/t/the-repository-https-packages-atlassian-com-atlassian-sdk-deb-stable-release-is-not-signed/36901/14

**Comment by raul-arabaolaza** [comment]
So, subversion and (I hope) jira plugin should be fixed on the last run, next step git plugin tests

**Comment by raul-arabaolaza** [comment]
`DeplyPluginTests` should be fixed now

**Comment by raul-arabaolaza** [comment]
From 228 skipped tests to 62, current failures look like unrelated or flaky to me, I am retriggering a build to be sure

@timja Do you remember those JobConfigHistory tests to be flaky?

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/Tomcat7Container/Dockerfile
Comment (dduportal) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/Tomcat7Container/Dockerfile:7]: That change worries me: https://releases.ubuntu.com/16.04/ is not maintained since 2019 (and 18.04 is almost end of life).

Can we (jenkins-infra) help on the packages versions part?

**Comment by raul-arabaolaza** [comment]
Yep, green build now

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/Tomcat7Container/Dockerfile
Comment (raul-arabaolaza) [reply src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/Tomcat7Container/Dockerfile:7]: @dduportal I would be forever in your debt if you can take a look at the ssh issues on `GitPluginTest` when using 22.04 as base. I believe I can workaround the issue on the `Tomcat7Container` myself

**Comment by raul-arabaolaza** [comment]
I believe I know what happened on the git container -> `t seems this has happened for the ssh client in Ubuntu 22.04. The RSA public-private key pair is considered not safe any more.` from https://askubuntu.com/questions/1409105/ubuntu-22-04-ssh-the-rsa-key-isnt-working-since-upgrading-from-20-04

**Comment by raul-arabaolaza** [comment]
Tried to use newly generated keys but JSCH does not support them, I am NOT gonna remove jsch here, so I am gonna try to reenable rsa keys

**Comment by raul-arabaolaza** [comment]
So, I still can ssh -i into the docker container using the old keys but tests are failing anyway... @dduportal I am totally out of ideas on the git container

**Comment by raul-arabaolaza** [comment]
Let's see what is broken on the `DeployPluginTest` now

**Comment by raul-arabaolaza** [comment]
I bet for some reason now the container uses java11 and tomcat7 is not compatible

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile
Comment (jtnord) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile:6]: most likely the line 24 is incorrect

`RUN cat /home/git/unsafe.pub >> /home/git/.ssh/authorized_keys`
is likely creating the file using the default keymask  (infact the line 19 `mkdir -p /home/git/.ssh` is probably just as bad.

the `.ssh` folder should be `0700` *and* be owned by the `git` user (not root)

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/Tomcat7Container/Dockerfile
Comment (dduportal) [reply src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/Tomcat7Container/Dockerfile]: Gotta try, but I need to (re)learn how it works (expect some overhead ;) )

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile
Comment (dduportal) [reply src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile]: +1 with @jtnord . I'm not sure how this Java code works to spin up Docker container (and I'm too scared to try it), but I got the following updated Dockerfile here tested as below (as a first step) and passed through `hadolint` (which has very very good tips and actionnables, like `shellcheck`):

```Dockerfile
FROM ubuntu:22.04

SHELL ["/bin/bash", "-o", "pipefail", "-c"]

# install SSHD, Git and zip, always on the latest available version
# hadolint ignore=DL3008
RUN apt-get update \
    && apt-get install --yes --no-install-recommends \
    ca-certificates \
    openssh-server \
    git \
    zip \
    && rm -rf /var/lib/apt/lists/*

# create a git user and setup its .ssh dir (including public key)
COPY unsafe.pub  /home/git/.ssh/authorized_keys
RUN useradd git -d /home/git \
    && echo "git:git" | chpasswd \
    && mkdir -p /var/run/sshd \
    && chmod 0700 /home/git/.ssh \
    && chmod 0600 /home/git/.ssh/* \
    && chown -R git /home/git

# run SSHD in the foreground with error messages to stderr
ENTRYPOINT ["/usr/sbin/sshd"]
CMD ["-D","-e"]
```

```shell
$ pwd
<my workspace>/acceptance-test-harness/src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer

$ docker build -t sshd ./
# Success

$ docker run --rm --detach --name=sshd-test --publish 2222:22 sshd
# Run in background the container

$ ssh -i ./unsafe -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -p2222 git@localhost whoami
# ..
git
## It works
```

**Comment by raul-arabaolaza** [comment]
@timja @dduportal @jtnord Can you re review please? AFAICT the JobConfigHistoryTest is a flake, it has been working and failing randomly during the course of this PR while my changes should not affect it at all

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile
Comment (raul-arabaolaza) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile:33]: We need this for jsch to be able to sftp into the container, I tried with more modern algorithms for the keys but jsch did not supported them

File: src/test/java/plugins/DeployPluginTest.java
Comment (raul-arabaolaza) [inline src/test/java/plugins/DeployPluginTest.java:37]: I do not believe deploying in tomcat 10 vs deploying in tomcat 7 changes the intent of the test

**Comment by timja** [comment]
> @timja @dduportal @jtnord Can you re review please? AFAICT the JobConfigHistoryTest is a flake, it has been working and failing randomly during the course of this PR while my changes should not affect it at all

You could file an issue and disable it if you don’t have time to look at it.

**Review by timja — APPROVED** [review]

**Comment by raul-arabaolaza** [comment]
Issues created, see link, and test ignored

**Comment by raul-arabaolaza** [comment]
@timja Can you please reenable auto merge?

**Review by NotMyFault — APPROVED** [review]

**Review by dduportal — APPROVED** [review]
👏 🚀

**Comment by timja** [comment]
Caused https://github.com/jenkinsci/acceptance-test-harness/issues/1322

**Pull request merged**
